lib/views/request/request_view.dart

```diff
@@ -177,12 +252,92 @@ class _RequestViewState extends State<RequestView> {
                             Icons.copy,
                             size: 20,
                           ))),
-                  const Spacer(flex: 2),
+                  const Spacer(flex: 1),
                 ],
               ));
         });
   }
 
+  void showDialogForRequest() {
+    RadioButtonData? selectedValue;
+    showDialog(
+      context: context,
+      barrierDismissible: false,
+      builder: (context) => AlertDialog(
+        title: Text('Choose an option',
+            style: Theme.of(context).textTheme.bodyLarge!.apply(
+                  fontSizeFactor: 1.6,
+                )),
+        content: SizedBox(
+          height: MediaQuery.of(context).size.height * 0.25,
+          width: MediaQuery.of(context).size.width * 0.25,
+          child: StatefulBuilder(
+            builder:
+                (BuildContext context, void Function(void Function()) set) {
+              return Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: [
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      const Text('Create a btc address'),
+                      Radio<RadioButtonData>(
+                        activeColor: Colors.pinkAccent,
+                        groupValue: selectedValue,
+                        onChanged: (RadioButtonData? value) async {
+                          set(() {
+                            selectedValue = value!;
+                          });
+                          await newaddr();
+                          if (context.mounted) {
+                            _showBottomDialog(
+                                context: context,
+                                identifier: RadioButtonData.btcAddress);
+                          }
+                        },
+                        value: RadioButtonData.btcAddress,
+                      ),
+                    ],
+                  ),
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      const Text('Create an lightning invoice'),
+                      Radio<RadioButtonData>(
+                        activeColor: Colors.pinkAccent,
+                        groupValue: selectedValue,
+                        onChanged: (RadioButtonData? value) async {
+                          set(() {
+                            selectedValue = value!;
+                          });
+                          await generateInvoice();
+                          if (context.mounted) {
```

Comment:
mh this always present if start to be an antipattern, so if you want to use it every where consider to wrap this in a general method and show the bottom dialog only if the context is mounted

otherwise reading the code is really bad in this way